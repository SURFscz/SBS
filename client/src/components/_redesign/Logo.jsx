import React from "react";

import {ReactComponent as NotFoundIcon} from "../../icons/image-not-found.svg";
import {isEmpty} from "../../utils/Utils";
import {srcUrl} from "../../utils/Image";

export default function Logo({src, className = "", alt = ""}) {
    if (isEmpty(src)) {
        return <NotFoundIcon/>;
    }
    const urlSrc = srcUrl(src, "jpeg");
    return <img src={urlSrc} alt={alt} className={className}/>


}