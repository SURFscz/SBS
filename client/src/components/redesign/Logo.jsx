import React from "react";

import {ReactComponent as NotFoundIcon} from "../../icons/image-not-found.svg";

export default function Logo({src, className = "", alt = ""}) {
    return src ? <img src={`data:image/jpeg;base64,${src}`} alt={alt} className={className}/> : <NotFoundIcon/>;
}