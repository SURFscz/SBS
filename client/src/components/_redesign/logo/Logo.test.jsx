import React from "react";
import { render } from "@testing-library/react";
import Logo from "./Logo";

describe("Logo", () => {
    it("renders with valid src", () => {
        const {asFragment} = render(
            <Logo src="valid-image.jpg" alt="Test Logo" className="test-class"/>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
