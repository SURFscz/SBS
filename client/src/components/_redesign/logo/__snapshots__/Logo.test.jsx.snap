// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders with valid src 1`] = `
<DocumentFragment>
  <img
    alt="Test Logo"
    class="test-class"
    src="data:image/jpeg;base64,valid-image.jpg"
  />
</DocumentFragment>
`;
